app/src/renderer/components/common/TmPage.vue

```diff
@@ -87,7 +87,7 @@ export default {
       default: undefined
     },
     dataset: {
-      type: Array,
+      type: [Array, Object],
```

Comment:
i know we agreed to not permit this. however, this might be time for an exception _or_ please share a suggestion for how to avoid this.

Reply:
this needs to be an array or more then this needs to be refactored. see line 28

Reply:
proposal: change to "dataAvailable" and make it a boolean. checks are then performed on each page on the individual dataset

Reply:
Also cheaper to pass

Reply:
ok makes sense. do you guys think this should be done in this PR and not separately?

Reply:
let's do it in here. the change needs to happen in only 3 places and we don't pile up more dept.

Reply:
> in only 3 places

yeah. one line 😝